framework/include/materials/Material.h

```diff
@@ -30,6 +30,53 @@ class Material : public MaterialBase, public Coupleable, public MaterialProperty
 
   Material(const InputParameters & parameters);
 
+  /**
+   * Gets an element integer for the proper current element with a parameter of
+   * the object derived from this interface
+   * Note: This overrides the function in ElementIDInterface to assure derived materials
+   *       to call the functions in ElementIDInterface properly.
+   */
+  virtual const dof_id_type & getElementID(const std::string & id_parameter_name,
+                                           unsigned int comp = 0) const override
+  {
+    return _neighbor ? ElementIDInterface::getElementIDNeighbor(id_parameter_name, comp)
+                     : ElementIDInterface::getElementID(id_parameter_name, comp);
+  }
+  /**
+   * Directly calling this function is not needed for materials because the same material has
+   * three copies for element interior, element side and neighbor side.
+   */
+  virtual const dof_id_type & getElementIDNeighbor(const std::string & id_parameter_name,
+                                                   unsigned int comp = 0) const override
+  {
+    mooseError("Directly calling getElementIDNeighbor is not allowed for materials intead of "
+               "calling getElementID");
```

Comment:
"Directly calling 'getElementIDNeighbor' is not allowed for materials. Please call 'getElementID' instead."